build_system/src/test.rs

```diff
@@ -1030,18 +1073,30 @@ fn test_successful_rustc(env: &Env, args: &TestArg) -> Result<(), String> {
     test_rustc_inner(
         env,
         args,
-        prepare_files_callback_success("tests/failing-ui-tests.txt", "ui"),
+        prepare_files_callback_remove("tests/failing-ui-tests.txt", "ui"),
+        None,
         "ui",
     )?;
     test_rustc_inner(
         env,
         args,
-        prepare_files_callback_success("tests/failing-run-make-tests.txt", "run-make"),
+        prepare_files_callback_remove("tests/failing-run-make-tests.txt", "run-make"),
+        None,
         "run-make",
     )
 }
 
-fn prepare_files_callback_failing<'a>(
+fn test_failing_ui_pattern_tests(env: &Env, args: &TestArg) -> Result<(), String> {
+    test_rustc_inner(
+        env,
+        args,
+        prepare_files_callback_remove("tests/failing-ice-tests.txt", "ui"),
+        Some(Box::new(|path| should_remove_test(path).unwrap_or(false))),
```

Comment:
Can you please explain why this is necessary?

Reply:
>   Some(Box::new(|path| should_remove_test(path).unwrap_or(false))),

I guess you are talking of this, as stated on first comment
>@antoyo after ideating for hours on how to proceed and looking for various options like implementing completely new function for this, including simple bool param in test_rustc_inner for this, making global bool variable to include/exclude these tests, etc.
I thought best option would be to include dynamic boxed closure callback function as parameter to test_rustc_inner function, this would enable us to implement separate tests/ci jobs for other checks pattern in different test suites if we implement in future.

basically as we currently wanted tests from `should_remove_test()` function to run (i.e. those followed the error patterns), in future we may have different other test suites with other function like `should_remove_test()`, then this callback would help in generalization.
Another way could be to just make another callback function that runs those tests, or if you have something better in mind let me know.

Reply:
I'm still not sure I understand.
We have the file `failing-ice-tests.txt` that tells which files to remove, so why do we need a callback in addition to this file?

Reply:
@antoyo could you elaborate on what do you mean by split the function? Do you want a separate function for `test_failing_ui_pattern_tests` having the funcitonality of `should_run_test_callback` in it?

Reply:
Actually, let's not split it since it seems complicated and use a bool param (or an enum if you think it's clearer). Then, add a comment to the new part (failing ui pattern tests) explaining that this remove the tests without an error pattern.

Reply:
Actually I also implemented it with bool initially but I feel this way though a little complicated would help in generalization in future with more test suits being integrated and more functions like `should_remove_tests` being involved. Let me implement the changes with proper comments, then let me see if you still want changes here

Reply:
Yeah, I prefer with the bool/enum: we can generalize later if there's really a need.